client/src/components/App.jsx

```diff
@@ -12,39 +12,90 @@ class App extends React.Component {
     super(props);
     this.state = {
       restaurantData: {},
-      // currentCategory: '',
+      currentSubMenu: '',
+      currentCollapseToggle: false,
       // menu : '',
       // menuHeaders: [],
     };
+    this.handleMenuCategoryClick = this.handleMenuCategoryClick.bind(this);
+    this.handleCollapseButtonClick = this.handleCollapseButtonClick.bind(this);
   }
 
   componentDidMount() {
-    axios.get('http://localhost:3002/api/menuData')
+    axios.get('http://localhost:3002/api/menuData/')
       .then((res) => {
         const restaurantsData = res.data;
-        this.setState({ restaurantData: restaurantsData[0] });
+        this.setState({
+          restaurantData: restaurantsData[0],
+          currentSubMenu: restaurantsData[0].menu_category[0].submenu_name,
+        });
       });
   }
 
+  handleMenuCategoryClick(e) {
+    console.log(e.target.id);
+    this.setState({ currentSubMenu: e.target.id });
+  }
+
+  handleCollapseButtonClick() {
+    const { currentCollapseToggle } = this.state;
+    const newToggle = !currentCollapseToggle;
+    this.setState({ currentCollapseToggle: newToggle });
+  }
+
   render() {
     const { restaurantData } = this.state;
+    const { currentSubMenu } = this.state;
+    const { currentCollapseToggle } = this.state;
```

Comment:
line 47 to 49, please combined to one line. 
```suggestion
    const { restaurantData , currentSubMenu , currentCollapseToggle } = this.state;
```

Reply:
Thanks this looks much cleaner